### ISSUE TITLE
Inconsistent lengths of the responses generated from the original image and the distorted image

### ISSUE THREAD

**Issue by Lookey-Luo** [DAMO-NLP-SG/VCD#35]
In Equation 3 of your paper, pvcd (y | v, v′, x) = softmax [(1 + α) logitθ (y | v, x) − α logitθ (y | v′, x)], where logitθ (y | v, x) represents the logit corresponding to the original input image, and logitθ (y | v′, x) represents the logit corresponding to the distorted image. However, when the lengths of the responses generated from the original image and the distorted image are inconsistent, how should we perform the subtraction? For example, if the response from the original image encounters the end token </s> at the 20th token, while the response from the distorted image is longer and has not yet appeared </s>, and needs to continue predicting subsequent tokens, how should the response optimized by the VCD algorithm be obtained? It is assumed that the response optimized by VCD is always shorter than both the original response and the distorted response, so this problem no longer exists. But the response optimized by VCD cannot always be the shortest, right?

**Comment by Vc-86** [comment]
in each step, the inputs_id are shared in two branches in VCD , we use two branches to get next-token, including the end_token.
